File: runtime/botmanager/src/com/newbound/code/primitive/object/Insert.java
package com.newbound.code.primitive.object;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import com.newbound.code.primitive.Primitive;

public class Insert extends Primitive 
{
	public Insert() throws JSONException {
		super("{ in: { a: {}, b: {} }, out: { c: {} } }");
	}
	
	@Override
	public JSONObject execute(JSONObject in) 
	{
		JSONObject out = new JSONObject();
		try
		{
			JSONArray a = in.getJSONArray("a");
			Object b = in.get("b");
			Object c = a.put(b);
			out.put("c", c);
		}
		catch (Exception x) { x.printStackTrace(); }

		return out;
	}
}

File: runtime/botmanager/src/newbound/code/primitive/object/insert.py
from ..primitive import Primitive


class Insert(Primitive):

    def __init__(self):
        super().__init__('{"in": {"a": {}, "b", {}}, "out": { "c": {} } }')

    def execute(self, input):
        out = {}
        a = input.get("a")
        b = input.get("b")
        a.append(b)
        c = a
        out["c"] = c
        return out
